server/jwt.go

```diff
@@ -17,7 +19,7 @@ import (
 
 const (
 	MicrosoftOnlineJWKSURL = "https://login.microsoftonline.com/common/discovery/v2.0/keys"
-	ExpectedAudience       = "api://community.mattermost.com/4ef56ea2-4a2f-4817-a6e0-a7cd760e2034"
```

Comment:
> The audience validation was fixed to community, let's format the URI so it can match wathever instance this plugin is installed into.

This is by design, as in a production deployment we expect this to match the Teams app we've deployed and no other. There is a case for it being paratemerizable, but only for customers who deploy this plugin and want to use the app they've deployed for their org. When we support that flow, we can make this whole thing an explicit configuration, but for now it has to be constant.

(Note that in developer mode, this audience check is skipped, so it's still possible to iterate without running the full suite of checks.)

Reply:
But for the deployed community plugin, this value will match the configuration, so it doesn't need to be fixed in the code, right?

Reply:
No, community is just passing the JWT to the downstream server which runs this code, so the SiteURL and ClientId wouldn't match. This has to be a fixed constant.

Reply:
Then maybe a setting in the plugin to enable this "routing behaviour" and check against that, but allow this for people that want to install their own manifest? cc: @wiggin77

Reply:
The [devsecops](https://github.com/mattermost/mattermost-plugin-msteams-devsecops) variant of this plugin will handle the non-routing customers, and help customers generate a manifest that can be installed. 

The [msteams](https://github.com/mattermost/mattermost-plugin-msteams) variant will (temporarily) handle the customers that need routing.  We will move the routing code to something like an Azure Function (lambda) once we have the mechanics nailed down.

Reply:
👍 

[feat: store user metadata in the plugin kvstore](https://github.com/mattermost/mattermost-plugin-msteams/pull/815/commits/46d3b7569c6e82a5f3816d508f66a43690d3db7f)